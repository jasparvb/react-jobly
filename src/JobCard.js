import React from "react";

function JobCard({title, salary, equity, applied, handleApply}) {
    return (
        <div className="Card card">
            <div className="card-body">
                <h6 className="card-title d-flex justify-content-between">
                    <span className="text-capitalize">{title}</span>
                </h6>
                <div>Salary: {salary}</div>
                <div>Equity: {equity}</div>
                <button className="btn btn-danger font-weight-bold text-uppercase float-right" 
                    disabled={applied}
                    onClick={handleApply}
                    >
                    {applied ? "Applied" : "Apply"}
                </button>
            </div>
        </div>
    );
}

export default JobCard;