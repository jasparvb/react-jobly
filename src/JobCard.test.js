import React from "react";
import { render } from "@testing-library/react";
import JobCard from "./JobCard";
import { MemoryRouter } from "react-router";

it("renders without crashing", function() {
  render(
    <MemoryRouter>
      <JobCard />
    </MemoryRouter>
  );
});

it("matches snapshot for job", function() {
  const { asFragment } = render(
    <MemoryRouter>
      <JobCard title="Software Engineer" salary="150000" equity="5" />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
