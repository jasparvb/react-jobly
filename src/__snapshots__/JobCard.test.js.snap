// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for job 1`] = `
<DocumentFragment>
  <div
    class="Card card"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title d-flex justify-content-between"
      >
        <span
          class="text-capitalize"
        >
          Software Engineer
        </span>
      </h6>
      <div>
        Salary: 150000
      </div>
      <div>
        Equity: 5
      </div>
      <button
        class="btn btn-danger font-weight-bold text-uppercase float-right"
      >
        Apply
      </button>
    </div>
  </div>
</DocumentFragment>
`;
